server/chat-plugins/chatlog.ts

```diff
@@ -353,24 +557,41 @@ export const pages: PageTable = {
 		}
 
 		void accessLog.writeLine(`${user.id}: <${roomid}> ${date}`);
-
 		this.title = '[Logs] ' + roomid;
 
-		if (date) {
+		const hasSearch = opts?.includes('search-') || opts?.includes('csearch-');
+		const search = opts?.slice(7).replace('-', '');
+		const isAll = (toID(date) === 'all' || toID(date) === 'alltime');
+
+		const parsedDate = new Date(date as string);
+		// this is apparently the best way to tell if a date is invalid
+		if (isNaN(parsedDate.getTime()) && !isAll && date !== 'today') {
+			return LogViewer.error(`Invalid date.`);
+		}
+
+		if (date && !hasSearch) {
 			if (date === 'today') {
 				return LogViewer.day(roomid, LogReader.today(), opts);
-			}
-			const parsedDate = new Date(date);
-			// this is apparently the best way to tell if a date is invalid
-			if (isNaN(parsedDate.getTime())) return LogViewer.error(`Invalid date.`);
-
-			if (date.split('-').length === 3) {
+			} else if (date.split('-').length === 3) {
 				return LogViewer.day(roomid, parsedDate.toISOString().slice(0, 10), opts);
 			} else {
 				return LogViewer.month(roomid, parsedDate.toISOString().slice(0, 7));
 			}
+		} else if (date && hasSearch && search) {
+			this.title = `[Search] [${room}] ${search}`;
+			if (Config.chatlogreader === 'fs') {
+				return LogSearcher.fsSearch(roomid, search, date, toID(cap));
+			} else if (Config.chatlogreader === 'ripgrep') {
+				return LogSearcher.ripgrepSearch(roomid, search, cap);
+			} else {
+				return LogViewer.error(
+					`<strong>Log searching has been configured incorrectly.<br>` +
+					`Please set Config.chatlogreader to 'fs' or 'ripgrep'.</strong>	`
+				);
+			}
```

Comment:
Maybe just default it to fs?

I'd also recommend throwing errors instead of using a more local error report. This ensures that the server owners are notified about the problem (rather than just the person using the command, who probably doesn't have the necessary access to fix it).

```suggestion
			if (Config.chatlogreader === 'fs' || !Config.chatlogreader) {
				return LogSearcher.fsSearch(roomid, search, date, toID(cap));
			} else if (Config.chatlogreader === 'ripgrep') {
				return LogSearcher.ripgrepSearch(roomid, search, cap);
			} else {
				throw new Error(`Config.chatlogreader must be 'fs' or 'ripgrep'.`);
			}
```